src/verilog/verilog_expr.cpp

```diff
@@ -533,11 +533,16 @@ exprt verilog_inside_exprt::lower() const
   return disjunction(disjuncts);
 }
 
-exprt verilog_past_exprt::default_value() const
+exprt verilog_past_exprt::default_value(const typet &type)
 {
-  auto zero = from_integer(0, type());
-  if(zero.is_nil())
-    throw "failed to create $past default value";
+  if(type.id() == ID_array)
+  {
+    auto &array_type = to_array_type(type);
+    auto zero_rec = default_value(array_type.element_type());
+    return array_of_exprt{zero_rec, array_type};
+  }
   else
-    return std::move(zero);
+  {
+    return from_integer(0, type);
+  }
 }
```

Comment:
Shouldn't this use CBMC's `zero_initializer` or else the one for Verilog being proposed in #1001?

Reply:
All done